ui_test_app/index.js

```diff
@@ -0,0 +1,88 @@
+const express = require("express");
+const fs = require("node:fs").promises;
+const { spawn } = require("node:child_process");
+
+class Streamsync {
+	constructor() {
+		this.process = null;
+	}
+
+	async start() {
+		return new Promise((resolve) => {
+			const ss = spawn(
+				"streamsync",
+				["edit", "./runtime", "--port", "7357"],
+				{
+					shell: true,
+				},
+			);
+			this.process = ss;
+
+			ss.stdout.on("data", (data) => {
+				// eslint-disable-next-line no-console
+				console.log(
+					`[${ss.pid}] stdout: ${Buffer.from(data, "utf-8").toString()}`,
+				);
+				if (data.includes("Builder is available at")) {
+					resolve(ss);
+				}
+			});
+
+			ss.stderr.on("data", (data) => {
+				// eslint-disable-next-line no-console
+				console.error(`[${ss.pid}] stderr: ${data}`);
+			});
+
+			ss.on("close", (code) => {
+				// eslint-disable-next-line no-console
+				console.log(
+					`[${ss.pid}] child process exited with code ${code}`,
+				);
+			});
+		});
+	}
+
+	async stop() {
+		return new Promise((resolve) => {
+			if (this.process) {
+				this.process.once("exit", () => {
+					this.process = null;
+					resolve();
+				});
+				this.process.kill("SIGINT");
+			} else {
+				resolve();
+			}
+		});
+	}
+
+	async restart() {
+		await this.stop();
+		await this.start();
+	}
+
+	async loadPreset(preset) {
+		await this.stop();
+		await fs.copyFile(`./presets/${preset}/ui.json`, "./runtime/ui.json");
+		await fs.copyFile(`./presets/${preset}/main.py`, "./runtime/main.py");
+		await this.start();
+	}
+}
+
+const ss = new Streamsync();
+(async () => {
+	await fs.mkdir("runtime", { recursive: true });
+	await ss.loadPreset("base");
+})();
+const app = express();
```

Comment:
Is this the magic you were talking about? 🧙‍♂️

Reply:
yup, It's a wrapper for Streamsync for reloading ui configuration file. Tests live on the frontend side so i needed some way to communicate with it to be able to trigger reload.